TooltipDecorator/tests/TooltipDecorator-specs.js

```diff
@@ -23,4 +29,160 @@ describe('TooltipDecorator', () => {
 			expect(tooltip).not.toHaveStyle({'text-align': unexpected});
 		});
 	});
+
+	describe('TooltipDecorator', () => {
+		beforeEach(() => {
+			global.Element.prototype.getBoundingClientRect = jest.fn(() => {
+				return {
+					width: 2500,
+					height: 2000,
+					top: 1000,
+					left: 1000,
+					bottom: 0,
+					right: 0
+				};
+			});
+		});
+
+		test('should render a tooltip if hovered', async () => {
+			const tooltipText = 'Tooltip';
+			render(
+				<FloatingLayerController>
+					<TooltipButton tooltipDelay={0} tooltipText={tooltipText}>Label</TooltipButton>
+				</FloatingLayerController>
+			);
+
+			const button = screen.getByRole('button');
+			act(() => button.focus());
+			fireEvent.mouseOver(button);
+
+			await waitFor(() => {
+				expect(screen.getByText('Tooltip')).not.toBeNull();
+			});
+		});
+
+		test('should hide tooltip if not hovered', async () => {
+			const tooltipText = 'Tooltip';
+			render(
+				<FloatingLayerController>
+					<TooltipButton tooltipDelay={0} tooltipText={tooltipText}>Label</TooltipButton>
+				</FloatingLayerController>
+			);
+
+			const button = screen.getByRole('button');
+			act(() => button.focus());
+			fireEvent.mouseOver(button);
+
+			await waitFor(() => {
+				expect(screen.getByText('Tooltip')).not.toBeNull();
+			});
+
+			act(() => button.blur());
+			fireEvent.mouseOut(button);
+
+			await waitFor(() => {
+				expect(screen.queryByText('Tooltip')).toBeNull();
+			});
+		});
+
+		// This test throws 'Warning: Each child in a list should have a unique "key" prop.'
```

Comment:
try to mock the implementation of console.error to supress the warning and remove this comment

Reply:
Done